soc: egis_et171: add customize config value of fw size info

jackylee-go opened the pull request:
This PR allows applications to customize the firmware size information for binary header optionally.

Although the firmware size is calculated at link time, sometimes applications need to reserve a fixed space for it.

For example:
A fixed-size firmware means that its signature will be located in a specified position.
This reduces the complexity of merging flash image files with multiple partitions.

niedzwiecki-dawid commented on soc/egis/et171/start.S:
```diff
@@ -22,7 +22,11 @@ SECTION_FUNC(init, entry)
 	.word 0x53494745 /* "EGIS" */
 	.word CONFIG_SOC_EGIS_ET171_FIRMWARE_VERSION /* version */
 	.word CONFIG_FLASH_BASE_ADDRESS + CONFIG_FLASH_LOAD_OFFSET /* flash start address */
+#if CONFIG_SOC_EGIS_ET171_VERNDOR_FIRMWARE_SIZE
```
Please use `#ifdef CONFIG_SOC_EGIS_ET171_VERNDOR_FIRMWARE_SIZE > 0` to be super clear.

niedzwiecki-dawid commented on soc/egis/et171/Kconfig:
```diff
@@ -30,3 +30,9 @@ config SOC_EGIS_ET171_FIRMWARE_VERSION
+
+config SOC_EGIS_ET171_VERNDOR_FIRMWARE_SIZE
+	hex "Overwrite the firmware size info of binary header"
+	default 0x0 # 0 means unused
```
Add the comment something like "Unused if zero" in the help message.

jackylee-go commented on soc/egis/et171/Kconfig:
```diff
@@ -30,3 +30,9 @@ config SOC_EGIS_ET171_FIRMWARE_VERSION
 	  Firmware version info of bin header
+
+config SOC_EGIS_ET171_VERNDOR_FIRMWARE_SIZE
+	hex "Overwrite the firmware size info of binary header"
```
Updated

jackylee-go commented on soc/egis/et171/Kconfig:
```diff
@@ -30,3 +30,9 @@ config SOC_EGIS_ET171_FIRMWARE_VERSION
+	hex "Overwrite the firmware size info of binary header"
+	default 0x0 # 0 means unused
+	help
+	  Customize the firmware size info of binary header
```
Updated

Commit:
soc: egis_et171: add customize config value of fw size info

Allows applications to customize the firmware size information
for binary header optionally.

Signed-off-by: Jacky Lee <jacky.lee@egistec.com>
